packages/vgrammar-core/src/interactions/view-roam.ts

```diff
@@ -0,0 +1,228 @@
+        if (events[this.options.drag.updateTrigger]) {
+          events[this.options.drag.updateTrigger].push(this.handleRoamDragUpdate);
+        } else {
+          events[this.options.drag.updateTrigger] = this.handleRoamDragUpdate;
+        }
+      }
+
+      if (this.options.drag?.endTrigger) {
+        if (events[this.options.drag.endTrigger]) {
+          events[this.options.drag.endTrigger].push(this.handleRoamDragEnd);
+        } else {
+          events[this.options.drag.endTrigger] = this.handleRoamDragEnd;
+        }
+      }
+    }
+
+    return events;
+  }
+
+  handleRoamZoomStartInner = (e: InteractionEvent) => {
+    (this as unknown as IViewZoomMixin).formatZoomEvent(e);
+    if (!e || (this.options.shouldStart && !this.options.shouldStart(e))) {
+      return;
+    }
+
+    if (!this._inited) {
+      this._initGrammars();
+    }
+
+    this.updateView(
+      InteractionEventEnum.viewZoomStart,
+      (this as unknown as IViewZoomMixin).handleZoomStart(e, this._state, this.options.zoom),
+      e
+    );
+  };
+
+  handleRoamZoomEnd = (e: InteractionEvent) => {
+    (this as unknown as IViewZoomMixin).formatZoomEvent(e);
+    if (!e || (this.options.shouldEnd && !this.options.shouldEnd(e))) {
+      return;
+    }
+
+    this.updateView(
+      InteractionEventEnum.viewZoomEnd,
+      (this as unknown as IViewZoomMixin).handleZoomEnd(e, this._state, this.options.zoom),
+      e
+    );
+  };
+
+  handleRoamZoomReset = (e: InteractionEvent) => {
+    if (!e || (this.options.shouldReset && !this.options.shouldReset(e))) {
+      return;
+    }
+    this.updateView(
+      InteractionEventEnum.viewZoomReset,
+      (this as unknown as IViewZoomMixin).handleZoomReset(e, this._state, this.options.zoom),
+      e
+    );
+  };
+
+  handleRoamDragStart = (e: InteractionEvent) => {
+    if (!e || (this.options.shouldStart && !this.options.shouldStart(e))) {
+      return;
+    }
+
+    if (!this._inited) {
+      this._initGrammars();
+    }
+
+    this.updateView(
+      InteractionEventEnum.viewDragStart,
+      (this as unknown as ViewDragMixin).handleDragStart(e, this._state, this.options.drag),
+      e
+    );
+  };
+
+  handleRoamDragUpdateInner = (e: InteractionEvent) => {
+    if (!e || (this.options.shouldUpdate && !this.options.shouldUpdate(e))) {
+      return;
+    }
+    this.updateView(
+      InteractionEventEnum.viewDragUpdate,
+      (this as unknown as ViewDragMixin).handleDragUpdate(e, this._state, this.options.drag),
+      e
+    );
+  };
+
+  handleRoamDragEnd = (e: InteractionEvent) => {
+    if (!e || (this.options.shouldEnd && !this.options.shouldEnd(e))) {
+      return;
+    }
+
+    this.updateView(
+      InteractionEventEnum.viewDragEnd,
+      (this as unknown as ViewDragMixin).handleDragEnd(e, this._state, this.options.drag),
+      e
+    );
+  };
+
+  handleRoamScrollStartInner = (e: InteractionEvent) => {
+    (this as unknown as IViewScrollMixin).formatScrollEvent(e);
```

Comment:
maybe use interface extend instead of unknown type conversion, so that the child class does not need to use unknown? `export interface ViewRoam extends IViewScrollMixin {}`